### ISSUE TITLE
Monzo API HTTP 401 response didn't trigger token refreshing

### ISSUE THREAD

**Issue by bartonp** [pawelad/pymonzo#6]
There was a use case where the reauth wasn't doing the required actions. I added my own checks in to see what was happening. The response code from the server was 401 which means you are unauthorised. Upon this error, it was also failing to re-authenticate. The error codes can be seen in the documentation here - https://monzo.com/docs/#errors.

If I run across the error again, I'll get some more debugging information added in to see what is actually happening in the long term.

Minimal code snippet below to see when it was happening:
```python
from ConfigParser import ConfigParser
import pymonzo

conf = ConfigParser()
conf.read('api_info.conf')
data = dict(conf.items('monzo'))

api = pymonzo.MonzoAPI(client_id=data['client_id'], client_secret=data['client_secret'])
```

**Comment by pawelad** [comment]
Looking at the source code of [`requests-oauthlib`](https://github.com/requests/requests-oauthlib) and [`oauthlib`](https://github.com/idan/oauthlib) I think you may be right and in some cases Monzo API could return 401 without `TokenExpiredError` being raised.

**Comment by pawelad** [comment]
Related: https://github.com/pawelad/pymonzo/issues/5#issuecomment-331651219

**Comment by pawelad** [comment]
This should be resolved in [v0.10.1](https://github.com/pawelad/pymonzo/releases/tag/v0.10.1)
